ios/brave-ios/Sources/Shared/URIFixup.swift

```diff
@@ -79,10 +85,56 @@ public class URIFixup {
     // the official URI scheme list, so that other such search phrases
     // like "filetype:" are recognised as searches rather than URLs.
     // Use `URL(string: entry)` so it doesn't double percent escape URLs.
-    if let url = entryURL, url.schemeIsValid {
+    if let url = entryURL as URL?, url.schemeIsValid {
+      // URLs containing a "user" are passed to the search engine
```

Comment:
why would we pass a url with a username to the search engine?

Reply:
This is what Chrome does and that's what Safari does.

Reply:
Fixed and removed. It turns out Chrome sends `user@brave.com` to the search engine, but it does not send: `user@brave.com/test` to the search engine.

Then on top of all of that, WebKit strips the `user` part of the domain, and only loads `brave.com` even if you typed `user@brave.com`.

This wasn't always the case in the past, but it seems like they fixed the security flaw, so we're all good :)